src/adapter/src/coord/sequencer.rs

```diff
@@ -518,6 +520,65 @@ impl Coordinator {
         }
     }
 
+    #[tracing::instrument(level = "debug", skip(self))]
+    pub(crate) async fn sequence_execute_single_statement_transaction(
+        &mut self,
+        ctx: ExecuteContext,
+        stmt: Statement<Raw>,
+        params: Params,
+    ) {
+        // Put the session into single statement implicit so anything can execute.
+        let (tx, internal_cmd_tx, mut session, extra) = ctx.into_parts();
+        assert!(matches!(session.transaction(), TransactionStatus::Default));
+        session.start_transaction_implicit(self.now_datetime(), 1);
```

Comment:
I think this is the functional call that confused me. Can we add a different function that only ever enters the `Started` state? Something like `start_transaction_single_stmt`.

Reply:
Done.